app/src/main-process/squirrel-updater.ts

```diff
@@ -1,34 +1,236 @@
+): Promise<void> {
+  await spawn(updateDotExe, commands)
+}
+
+type ShortcutLocations = ReadonlyArray<'StartMenu' | 'Desktop'>
+
+function createShortcut(locations: ShortcutLocations): Promise<void> {
+  return spawnSquirrelUpdate([
+    '--createShortcut',
+    exeName,
+    '-l',
+    locations.join(','),
+  ])
+}
+
+async function handleUninstall(): Promise<void> {
+  await removeShortcut()
+
+  const paths = await getPathSegments()
+  const binPath = getBinPath()
+  const pathsWithoutBinPath = paths.filter(p => p !== binPath)
+  return setPathSegments(pathsWithoutBinPath)
+}
+
+function removeShortcut(): Promise<void> {
+  return spawnSquirrelUpdate(['--removeShortcut', exeName])
+}
+
+function updateShortcut(): Promise<void> {
+  const homeDirectory = Os.homedir()
+  if (homeDirectory) {
+    const desktopShortcutPath = Path.join(
+      homeDirectory,
+      'Desktop',
+      'GitHub Desktop.lnk'
+    )
+    return new Promise<void>((resolve, reject) => {
+      Fs.exists(desktopShortcutPath, exists => {
+        const locations: ShortcutLocations = exists
+          ? ['StartMenu', 'Desktop']
+          : ['StartMenu']
+        createShortcut(locations).then(resolve).catch(reject)
+      })
+    })
+  } else {
+    return createShortcut(['StartMenu', 'Desktop'])
+  }
+}
+
+/** Get the path segments in the user's `Path`. */
+async function getPathSegments(): Promise<ReadonlyArray<string>> {
+  let powershellPath: string
+  const systemRoot = process.env['SystemRoot']
+  if (systemRoot) {
+    const system32Path = Path.join(process.env.SystemRoot, 'System32')
+    powershellPath = Path.join(
+      system32Path,
+      'WindowsPowerShell',
+      'v1.0',
+      'powershell.exe'
+    )
+  } else {
+    powershellPath = 'powershell.exe'
   }
 
-  return false
+  const args = [
+    '-noprofile',
+    '-ExecutionPolicy',
+    'RemoteSigned',
+    '-command',
+    // Set encoding and execute the command, capture the output, and return it
+    // via .NET's console in order to have consistent UTF-8 encoding.
+    // See http://stackoverflow.com/questions/22349139/utf-8-output-from-powershell
+    // to address https://github.com/atom/atom/issues/5063
+    `
+      [Console]::OutputEncoding=[System.Text.Encoding]::UTF8
+      $output=[environment]::GetEnvironmentVariable('Path', 'User')
+      [Console]::WriteLine($output)
+    `,
+  ]
+
+  const stdout = await spawn(powershellPath, args)
+  const pathOutput = stdout.replace(/^\s+|\s+$/g, '')
+  return pathOutput.split(/;+/).filter(segment => segment.length)
+}
+
+/** Set the user's `Path`. */
+async function setPathSegments(paths: ReadonlyArray<string>): Promise<void> {
+  let setxPath: string
+  const systemRoot = process.env['SystemRoot']
+  if (systemRoot) {
+    const system32Path = Path.join(systemRoot, 'System32')
+    setxPath = Path.join(system32Path, 'setx.exe')
+  } else {
+    setxPath = 'setx.exe'
+  }
+
+  await spawn(setxPath, ['Path', paths.join(';')])
```

Comment:
Ah, yeah, so the way Squirrel events work is you're meant to do some action and then quit ASAP. It's not a normal run of the app. So it shouldn't matter.